canonical-web-and-design/maas-ui
# fix(ui): Fix resizing cores chart bug in KVM details page when viewed on Firefox.

## Done

- Removed some leftover code that handled explicitly setting segmented bar chart width. It was no longer necessary since the chart width is now fluid instead of locked to a base-2 number. I'm not 100% sure why the bug only showed on firefox, but I suspect it has something to do with how CSS grid handles widths, because the bug was only present in KVM details (where it's in a grid) and not the KVM list (where it's in a table).

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the KVM details page in Firefox.
- Resize the window and check that the cores chart correctly resizes the width and the individual segments.
- Do the same on the KVM list.

## Fixes

Fixes #1580
